src/main/java/com/google/sps/servlets/Caption.java

```diff
@@ -23,15 +29,26 @@
   public String getCaptionFromId(String videoId) {
     // BCP47 language tags
     final String[] languages = {"en", "en-US", "en-GB", "en-AU", "en-IE", "en-ZA"};
-    for (String language : languages) {
-      try {
-        return parseXmlFromStream(captionService.fetchStream(videoId, language));
-      } catch (SAXException | ParserConfigurationException | IOException
-          | IllegalArgumentException e) {
-        continue;
-      }
+
+    ExecutorService executor = Executors.newFixedThreadPool(languages.length);
+
+    List<Callable<String>> captionCallables =
+        Arrays.stream(languages).map(language -> new Callable<String>() {
+          @Override
+          public String call() throws Exception {
+            return parseXmlFromStream(captionService.fetchStream(videoId, language));
+          }
+        }).collect(Collectors.toList());
+
+    String caption = "";
+    try {
+      caption = executor.invokeAny(captionCallables);
```

Comment:
Now that I think about it, does the sentiment/perspective API determine the text language is and then calculates a score?

Reply:
@kayefung says it the sentiment API detects the language before analyzing. Not sure about perspective API though. The "different" languages here are variants of English.